src/js/hooks/useArrowsNavigation.jsx

```diff
@@ -66,25 +68,66 @@ const useArrowsNavigation = ({
   };
 
   const handleKeyDown = (e, rowIndex, columnId) => {
-    const { key } = e;
+    const { key, ctrlKey } = e;
 
     if (key === navigationKey.ARROW_UP) {
       const isInArray = existingRowFocusableCells.includes(columnId);
       if (rowIndex > 0 && (isInArray || isNewRow(tableData[rowIndex - 1]))) {
         setRowIndex(rowIndex - 1);
         setColumnId(columnId);
+        onBlur();
       }
-      onBlur();
       e.preventDefault();
       e.stopPropagation();
     }
 
     if (key === navigationKey.ARROW_DOWN) {
-      if (rowIndex < tableData.length - 1) {
+      const isLastRow = rowIndex === tableData.length - 1;
+      const isLastColumn = columnId
+        === existingRowFocusableCells[existingRowFocusableCells.length - 1];
+
+      // Here we check getValues and setValue because in cycleCount I haven't implemented it yet.
+      // It's a safeguard to ensure they are available before using them
+      if (ctrlKey && !isLastRow && getValues && setValue) {
+        const currentValue = getValues(`values.lineItems.${rowIndex}.${columnId}`);
+        const nextRowIndex = rowIndex + 1;
+        setRowIndex(nextRowIndex);
+        setColumnId(columnId);
+        // Only copy the value if it's not empty/undefined/null
+        if (currentValue !== '' && currentValue !== undefined && currentValue !== null) {
+          setValue(`values.lineItems.${nextRowIndex}.${columnId}`, currentValue);
+        }
+        onBlur();
+        // We add e.preventDefault(), e.stopPropagation and return here
+        // to stop further event propagation, as the next condition below
+        // might also be met and add another row unintentionally
+        e.preventDefault();
+        e.stopPropagation();
+        return;
+      }
+
+      if (!ctrlKey && isLastRow && isLastColumn && getValues && setValue) {
+        addNewRow();
+        setRowIndex(rowIndex + 1);
+        setColumnId(newRowFocusableCells[0]);
+        onBlur();
+        e.preventDefault();
+        e.stopPropagation();
+        return;
+      }
+      if (!ctrlKey && isLastRow && isLastColumn) {
+        addNewRow();
+        setRowIndex(rowIndex + 1);
+        setColumnId(newRowFocusableCells[0]);
+        onBlur();
+      }
+
+      if (!isLastRow) {
```

Comment:
I would love to split this logic into new functions, like handleXYZ, and call it in a specific `if` - it will be more descriptive